servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/NettyIoThreadFactory.java

```diff
@@ -29,7 +26,7 @@
 /**
  * Default {@link IoThreadFactory} to create IO {@link NettyIoThread}s.
  */
-public final class NettyIoThreadFactory implements IoThreadFactory<NettyIoThreadFactory.NettyIoThread> {
+public final class NettyIoThreadFactory implements IoThreadFactory<NettyIoThread> {
```

Comment:
maybe awkward to have a non-public type for the generic argument but we don't actually need to expose this type.

Reply:
if it passes all tests 🚀